hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+        },
+        { bits: "7:0",
+          name: "MANUAL_CS"
+          desc: '''When MANCS_EN = 1, the cs_n lines are directly to the
+                   value of MANUAL_CS.'''
+          resval: 0
+        }
+      ]
+    },
+    { name: "STATUS",
+      desc: "Status register",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "31",
+          name: "READY",
+          desc: '''When high, indicates the SPI host is ready to receive
+                   commands. Writing to SPICMD when RDY is low is an error,
+                   and will trigger an interrupt.
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "30",
+          name: "ACTIVE",
+          desc: '''When high, indicates the SPI host is processing a previously
+                   issued command.'''
+          resval: "0x0"
+        },
+        { bits: "29",
+          name: "TXFULL",
+          desc: '''When high, indicates that the transmit data fifo is full.
+                   Any further writes to TXD will create an error interrupt.
+                '''
+          resval: "0x0"
+        },
+        { bits: "28",
+          name: "TXEMPTY",
+          desc: '''When high, indicates that the transmit data fifo is empty.
+                   And pending commands may have to wait for more data.
+                '''
+          resval: "0x0"
+        },
+        { bits: "27"
+          name: "TXSTALL",
+          desc: '''If high, signifies that an ongoing transaction has stalled
+                   due to lack of data in the TX FIFO''',
+          resval: "0x0"
+        },
+        { bits: "26",
+          name: "TXWM",
+          desc: '''If high, the amount of data in the TX FIFO has fallen below the
+                   level of CONFIG.TX_WATERMARK words (32b each). To prevent the
+                   reassertion of this interrupt add more data ot the TXFIFO, or
+                   reduce CONFIG.TX_WATERMARK.''',
+          resval: "0x0"
+        },
+        { bits: "25"
+          name: "RXFULL",
+          desc: '''When high, indicates that the receive fifo is full.  Any
+                   ongoing transactions will stall until firmware reads some
+                   from the RXFIFO.'''
+          resval: "0x0"
+        },
+        { bits: "24"
+          name: "RXEMPTY",
+          desc: '''When high, indicates that the receive fifo is empty.
+                   Any reads from RXFIFO will cause an error interrupt.
+                   '''
+          resval: "0x0"
+        },
+        { bits: "23"
+          name: "RXSTALL",
+          desc: '''If high, signifies that an ongoing transaction has stalled
+                   due to lack of available space in the RX FIFO''',
+          resval: "0x0"
+        },
+        { bits: "20",
+          name: "RXWM",
+          desc: '''If high, the amount of data in the RX FIFO now exceeds the
+                   level of CONFIG.RX_WATERMARK words (32b each). To prevent the
```

Comment:
Maybe this could be "... the number of 32-bit words in the RX FIFO now exceeds CONFIG.RX_WATERMARK" ? I think it's a bit easier to read and we could even replace "32-bit words" with "entries": the reader hopefully knows the width of the FIFO :-)

Also, maybe the note about reassertion of this interrupt should move to EVENT_ENABLE, since the STATUS register doesn't isn't really about an interrupt.

The same applies to TXWM, of course.

Reply:
Good Suggestion!